src/arviz_plots/plots/tracedensplot.py

```diff
@@ -0,0 +1,241 @@
+    plot_kwargs.setdefault("trace", {}).setdefault("lw", linewidth)
+
+    if plot_collection is None:
+        if backend is None:
+            backend = rcParams["plot.backend"]
+        pc_kwargs.setdefault("cols", ["__column__"])
+        pc_kwargs.setdefault("plot_grid_kws", {"figsize": figsize})
+
+        if compact:
+            pc_kwargs.setdefault("rows", ["__variable__"])
+        else:
+            pc_kwargs.setdefault("rows", ["__variable__"] + aux_dim_list)
+
+        plot_collection = PlotCollection.grid(
+            posterior.expand_dims(__column__=2),
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    if aes_map is None:
+        aes_map = {"trace": plot_collection.aes_set}
+
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    density_dims, _, density_ignore = filter_aes(plot_collection, aes_map, "trace", sample_dims)
+
+    if compact:
+        density_dims += aux_dim_list
+
+    # dens
+    if kind == "kde":
+        density = posterior.azstats.kde(dims=density_dims, **dens_kwargs.get("density", {}))
+        plot_collection.map(
+            line_xy,
+            "dens",
+            data=density,
+            ignore_aes=density_ignore,
+            coords={"__column__": 0},
+            **plot_kwargs.get("dens", {}),
+        )
+
+    elif kind == "ecdf":
+        density = posterior.azstats.ecdf(dims=density_dims, **dens_kwargs.get("density", {}))
+        plot_collection.map(
+            ecdf_line,
+            "dens",
+            data=density,
+            ignore_aes=density_ignore,
+            coords={"__column__": 0},
+            **plot_kwargs.get("dens", {}),
+        )
+
+    # trace
+    plot_collection.map(
+        line,
+        "trace",
+        data=posterior,
+        ignore_aes=density_ignore,
+        coords={"__column__": 1},
+        **plot_kwargs.get("trace", {}),
+    )
+
+    ## aesthetics
+    # Remove yticks, only for KDEs
+    if kind == "kde":
+        _, _, yticks_dens_ignore = filter_aes(plot_collection, aes_map, "yticks_dens", sample_dims)
+
+        plot_collection.map(
+            remove_ticks,
+            "yticks_dens",
+            ignore_aes=yticks_dens_ignore,
+            coords={"__column__": 0},
+        )
+
+    # Add varnames as x and y labels
+    _, labels_dens_aes, labels_dens_ignore = filter_aes(
+        plot_collection, aes_map, "labels_dens", sample_dims
+    )
+    labels_dens_kwargs = dens_kwargs.get("labels_dens", {}).copy()
+
+    if "color" not in labels_dens_aes:
+        labels_dens_kwargs.setdefault("color", "black")
+
+    labels_dens_kwargs.setdefault("fontsize", textsize)
+
+    plot_collection.map(
+        labelled_x,
+        "labels_dens",
+        ignore_aes=labels_dens_ignore,
+        coords={"__column__": 0},
+        subset_info=True,
+        labeller=labeller,
+        **labels_dens_kwargs,
+    )
+
+    plot_collection.map(
+        labelled_y,
+        "labels_dens",
```

Comment:
if this doesn't error out I should add a check to prevent repeated artist names.